Ma frontpage rebuild

micada opened the pull request:
# What does this pull request do?
Rebuilds each content/copy area in the front page to be updated via ACF
The front page is not as flexible as the other templates and so we're keeping it pretty tight with regard to the layout and markup.

# How should the reviewer test this pull request?
You should probably pull a staging DB down since that has the fields you need but you would have to test this locally

# Include the Trello card for this PR, if there is one.
https://trello.com/c/Y8npYlMF/5-updating-frontpage-for-wp-content

# Include any screenshots that will help explain this PR.
N/A

# Any another context you want to provide?
👺

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,98 +1,88 @@
             </div>
-          </div>
+            <?php
+            wp_reset_postdata();
```
This should go after the `endwhile` on the next line.

jpetto replied:
Looks like this same bug is present on line 124. Might as well fix in this PR.

micada replied:
These are addressed!

gensym commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,98 +1,88 @@
   </div>
   <main>
-    <section class="hero">
+    <section class="hero" style="background-image: url('<?php the_field('hero_image'); ?>');">
```
It's possible that I'm populating this incorrectly, but I was unable to get the hero image to show up on the page.

micada replied:
Yeah I had a namespace issue with the inner hero image ACF. Fixed!

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,98 +1,88 @@
   </div>
   <main>
-    <section class="hero">
+    <section class="hero" style="background-image: url('<?php the_field('hero_image'); ?>');">
```
Can probably remove [the legacy CSS that used to set the bg image](https://github.com/pawar-2018/pawar2018/blob/dev/wp-content/themes/pawar2018/sass/modules/_hero.scss#L2)?

micada replied:
Done!

gensym commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,98 +1,88 @@
         <div class="small-11 large-6 columns">
           <div class="flex-video widescreen">
-            <iframe width="500" height="282" src="https://www.youtube.com/embed/0QN_PvW9nQw?rel=0&amp;showinfo=0" frameborder="0" allowfullscreen></iframe>
+            <?php the_field('featured_video'); ?>
```
Should the contents of `featured_video` be just a link to the video URL? It appeared that way from the form, but after dropping a link to the youtube URL, it doesn't seem to be appearing.

micada replied:
Yep just a youtube video url. There were a couple weird things going on with this, but it should work now.

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,98 +1,88 @@
-              <p>
-                Pass criminal justice reform and refocus resources from prisons to diversion programs, job training and placement, and social and mental health services.
-              </p>
+          <?php $loop = new WP_Query( $args ); ?>
```
Where is `$args` defined?

micada replied:
I removed this

micada commented:
@jpetto @gensym I addressed most of your comments. Also Staging is set up with the correct information and if you pull down you should be able to get all of the pre-populated fields.

Finally I added eventbrite which doesn't work in locals because of how Keyring has to be authenticated. It's basically a straight copy/paste of what's in /events though.

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,188 +1,164 @@
+                      <span class="event-address">
+                        <?= eventbrite_event_venue()->address->localized_multi_line_address_display[0]; ?>
+                        <br/>
+                        <?= eventbrite_event_venue()->address->localized_multi_line_address_display[1]; ?>
```
Better to stick with convention and use `<?php echo` instead of the shortcut `<?=`, here and above.

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,188 +1,164 @@
+              <img src="<?php echo get_field('pillar_icon') ?>" class="issue-icon" alt="<?php echo get_field('pillar_alt') ?>">
+              <div class="pillar-copy">
+                <h4 class="pillar-header"><?php the_title();?></h4>
+                <p><?php the_content() ?></p>
```
`<?php the_content() ?>` generates wrapping `<p>` tags, so this ends up with double `<p>` tags. Might be better to do `<p><?php echo get_the_content() ?></p>`.

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,188 +1,164 @@
-                        <?php the_field('start_date'); ?> at <?php the_field('start_time'); ?>
-                    </p>
-                    <h5 class="event-title">
+            <?php $events = new Eventbrite_Query(
```
We can make this a little better for local dev by wrapping the entire thing in the following:

```

<?php
// eventbrite doesn't work locally :(
if (class_exists(Eventbrite_Query)) : ?>
    // all the stuff you have already
    wp_reset_postdata(); // you already have this - just repeating for placement
else :
    // maybe something more informative here?
    get_template_part('content', 'none');
endif;
?>
```

That will at least suppress the error message and allow the rest of the page to render.

jpetto commented on wp-content/themes/pawar2018/front-page.php:
```diff
@@ -1,188 +1,164 @@
+              while ($events->have_posts()) :
+                $events->the_post(); ?>
+                  <div class="column small-11 large-8 event-copy">
+                    <?php get_field('link') ?>
```
`get_field` doesn't display anything, so not sure what's intended here.
